config/config.go

```diff
@@ -849,6 +853,19 @@ func (c *ScrapeConfig) Validate(globalConfig GlobalConfig) error {
 		c.MetricNameValidationScheme = globalConfig.MetricNameValidationScheme
 	}
 
+	switch globalConfig.MetricNameEscapingScheme {
```

Comment:
Hm, isn't something odd here? Why we check global and only then apply optionally this global? Shouldn't we first apply global if `c.MetricNameEscapingScheme` is empty and THEN do this switch using  scrape level, eventual value of `c.MetricNameEscapingScheme`? 

Otherwise we don't validate correctly the following case: 
1. `globalConfig.MetricNameEscapingScheme: model.EscapeUnderscores`
2. `c.MetricNameEscapingScheme: model.AllowUTF8` and c.MetricNameValidationScheme = LegacyValidation

Reply:
If that's true and you want to fix it, please also fix `MetricNameValidationScheme` which has the same gap.

Reply:
We only want to apply the global if the local value is not set. However you're right that we do not validate the local setting.  The order is fine, though, I think?, I have an override test in config_test with scrape_config_local_global_validation_mode.

Reply:
We could optionally apply the default values here and not have the blank strings passed through

Reply:
With the changed code my statement is no longer true, looks fixed. thanks!